### DISCUSSION TITLE
Disk usage limit, was it there, is it there?

### DISCUSSION THREAD

**Discussion by Frono** [Nachtzuster/BirdNET-Pi#241] [Q&A] [answered]
I'm trying to remember there used to be a setting from long ago that set the disk usage limit for the audio recordings cleanup etc... But I do not see it in the setup menu. 
I have a large storage media for the birdnetpi, and would like to cut the size used for recordings so I can actually run the new backup script to create the backup file on the birdnetpi's SD card. That way I can (via the web interface, since I'm rarely near the location of the pi) backup the birdnetpi, with it's detections/reports etc, and download that backup all via the network connection without having to physically contact the birdnet pi...

I think the backup script is in place, but it tells me there is not enough room, I went back and just made the setting to only keep 10 of each birdsong, but would like to know if I was dreaming or if the disk usage limit really existed, and if it still exists... :)

**Comment by alexbelgium** [comment, marked as answer]
Hi, there is 2 options in the advanced tab to manage the disk. 

The first one is Purge/Keep : of you select purge then it will cap size at 95% and routinely eliminate everything that goes above. The 95% is not configurable.
https://github.com/Nachtzuster/BirdNET-Pi/blob/abcbcb601871f99c40ce757010a0772d767653be/scripts/disk_check.sh#L7

The second one is "Amount of files to keep for each species : ". If different than 0 (keep all), defines the maximum number of files to be kept for each species, with priority give to files with highest confidence. This value does not take into account the last 7 days (protected by default).
Note only the spectrogram and audio files are deleted, the obsevation data remains in the database. The files protected through the "lock" icon are also not affected

If the backup script tells you there is not enough place then one of the 2 options is likely not working and it's a bug : could you please check the disk size and the amount of files for a particular bird (can be through with a command like `find "$HOME/BirdSongs" -name *bird_species*mp3`)?

**Comment by Frono** [comment]
Worked like a champ! Thanks! From my memory at one time there was a config option to change that 95% to another number, I remember tweaking it, but it probably went away sometime during the last repository, as it is not in any recent configs...
However, just keeping 20 birdsongs made it really easy to compress a 128gig drive down to a 3 gig backup via the backup_data.sh shell command.
